jobserver/views/projects.py

```diff
@@ -188,17 +189,41 @@ def get_context_data(self, **kwargs):
 
         repos = sorted(set(workspaces.values_list("repo", flat=True)))
 
-        releases_count = Release.objects.filter(workspace__project=self.object).count()
-
         return super().get_context_data(**kwargs) | {
             "can_manage_workspaces": can_manage_workspaces,
             "can_manage_members": can_manage_members,
             "can_use_releases": can_use_releases,
-            "releases_count": releases_count,
+            "outputs": self.get_outputs(workspaces),
             "repos": list(self.get_repos(repos)),
             "workspaces": workspaces,
         }
 
+    def get_outputs(self, workspaces):
+        """
+        Builds up a QuerySet of Snapshots (outputs) for the page
+
+        We only want to show the most recent published Snapshot for each
+        Workspace.
+        """
+        # build a Subquery to get the PK for the most recently published
+        # Snapshot for each Workspace
+        latest_snapshot_qs = (
+            Snapshot.objects.filter(workspace_id=OuterRef("id"))
+            .exclude(published_at=None)
+            .order_by("-published_at")
+            .values("pk")[:1]
```

Comment:
OOI, I assume this ends up issuing a LIMIT 1 on the subquery?

Reply:
That's right, all slicing on a QuerySet becomes a `LIMIT` on the query.